src/sources/util/tcp.rs

```diff
@@ -115,17 +117,47 @@ pub trait TcpSource: Clone + Send + 'static {
 
                         let out = out.clone();
 
-                        let events_in = FramedRead::new(socket, source.decoder())
-                            .take_until(tripwire)
-                            .filter_map(move |frame| {
-                                let host = host.clone();
-                                source.build_event(frame, host)
-                            })
-                            .map_err(|error| warn!(message = "connection error.", %error));
-
-                        let handler = events_in.forward(out).map(|_| debug!("connection closed."));
-
-                        tokio::spawn(handler.instrument(span.clone()));
+                        if let Some(tls) = tls.clone() {
+                            match tls.acceptor() {
+                                Err(error) => error!(message = "Failed to create a TLS connection acceptor", %error),
+                                Ok(acceptor)=> {
+                                    let handler = TlsAcceptor::from(acceptor)
+                                        .accept(socket)
+                                        .map_err(
+                                            |error| warn!(message = "TLS connection accept error.", %error),
+                                        )
+                                        .map(|socket| {
+                                            let handler = FramedRead::new(socket, source.decoder())
+                                                .take_until(tripwire)
+                                                .filter_map(move |frame| {
+                                                    let host = host.clone();
+                                                    source.build_event(frame, host)
+                                                })
+                                                .map_err(
+                                                    |error| warn!(message = "connection error.", %error),
+                                                )
+                                                .forward(out)
+                                                .map(|_| debug!("TLS connection closed."));
+                                            tokio::spawn(handler);
```

Comment:
this has no span attached to it, we need to ensure it gets the correct span.

Reply:
Done